### PR TITLE
Move signal handing out of tracee-ebpf package

### LINKED ISSUE #975: Don't handle signals in tracee package, but in main
Today we handle the SIGINT and SIGTERM signals in the tracee package (of tracee-ebpf).
A user that wants to use tracee-ebpf as a library may want to set its own handlers, and also be able to gracefully close the perf buffers and save the captured artifacts as done in the Run() function (and not in the Close() function).
A better approach to signal tracee to close can be done by using go context, which will then enable the caller (either main or a user that is using tracee-ebpf as a library) to call the context's cancelation function, and close tracee-ebpf gracefully.

### PR DESCRIPTION
**Pull request by liamg**
Resolves #975

Other projects that wish to use tracee-ebpf as a library can now use the package with a `context.Context` to stop running: `Run() error` is now `Run(ctx context.Context) error`. Previously the `Run()` method could only be stopped using `SIGINT`/`SIGTERM`.

The tracee-ebpf binary now creates a context that is cancelled by these interrupts and passes it to the tracee-ebpf package so existing behaviour is preserved.

NOTE: I've imported `"context"` using the `gocontext` alias as `context` is already used elsewhere in the same package for another import. 

NOTE 2: This is my first tracee PR and I'm pretty unfamiliar with things atm, so feel free to rip this apart if necessary :)

### CODE REVIEW

**Comment by yanivagman** [comment]
Thanks @liamg !
This is definitely something that we need to take out of the tracee package into main.
@mtcherni95 I see that you assigned yourself for this task, do you mind taking a look at this PR?

**Comment by mtcherni95** [comment]
> Thanks @liamg !
> This is definitely something that we need to take out of the tracee package into main.
> @mtcherni95 I see that you assigned yourself for this task, do you mind taking a look at this PR?

no problem - I'll review :)

File: tracee-ebpf/tracee/tracee.go [hunk 1]
```diff
@@ -949,7 +947,7 @@ func (t *Tracee) Run() error {
 	go t.handleEvents(t.config.ChanDone)
 	go t.processFileWrites()
 	go t.processNetEvents()
-	<-sig
+	<-ctx.Done()
```
Comment (mtcherni95) [inline tracee-ebpf/tracee/tracee.go:950]: I would add here a comment stating that this operation is actually blocking.

**Review by mtcherni95 — COMMENTED** [review]
Looks good overall! I just gave a small comment. Please squash your commits into one.

**Review by yanivagman — COMMENTED** [review]
A different approach to pass context (instead of `Run(ctx)`) might be through Tracee config (which is given to `tracee.New()`).
Although this will change Tracee's config API, we can then remove `cfg.ChanDone` and pass ctx to the events pipeline instead. I think this approach is cleaner than the `ChanDone` approach we have today to terminate the pipeline (which requires to close the channel to signal program termination). WDYT @liamg @mtcherni95 ?

**Comment by liamg** [comment]
@mtcherni95 Thanks, I'll add the comment :+1: 

@yanivagman That sounds good, I'll take a look later on and see if I can do what you suggested :+1:

**Comment by liamg** [comment]
@yanivagman I've made that change - WDYT?

**Comment by mtcherni95** [comment]
> A different approach to pass context (instead of `Run(ctx)`) might be through Tracee config (which is given to `tracee.New()`). Although this will change Tracee's config API, we can then remove `cfg.ChanDone` and pass ctx to the events pipeline instead. I think this approach is cleaner than the `ChanDone` approach we have today to terminate the pipeline (which requires to close the channel to signal program termination). WDYT @liamg @mtcherni95 ?
> 
> **Edit: we might not even need to pass it through the config as the pipeline is started in Run(). So the comment is more about removing ChanDone from Tracee config and using ctx instead for the pipeline.**

yeah, that's a good idea.

**Comment by yanivagman** [comment]
> @yanivagman I've made that change - WDYT?

Thanks @liamg.
LGTM
Seems that a rebase is now required

**Comment by liamg** [comment]
@yanivagman Thanks, have rebased now :+1:

**Review by yanivagman — APPROVED** [review]

**Pull request merged**